### PR TITLE
Add export to Flipper Zero

### LINKED ISSUE #2: Convert to Flipper Zero  ?
Is there a way to convert dumped IR to Flipper Zero Format ???
thx :)

### PR DESCRIPTION
**Pull request by LionZXY**
Main goal of this pr is add new export type - `flipper` for Flipper Zero

But I have some problem with passing other keys from `load_brand_codes` - right now parser ignore the button which does not just power on/off the device

# Changes
- `load_brand_codes` right now return all keys which available for this brand
-  Add new export - `flipper`

# How to test
I tested:
- Full download (from zero)
- Partial download
- TVKill export
- Flipper Zero export

### CODE REVIEW

**Comment by LionZXY** [comment]
Fix for #2

**Comment by ysard** [comment]
Hi, thank you for your interest in the project. 

I will analyze your code and take note of the changes you have made.
I need some time to get back into this 2 years old project.

Just a 1st remark, next time, try to use more the `git add --patch` command (I never use `add` alone).
It allows you to make minimalist commits focused on a specific change.
It allows you to work on many things in a file without having to commit all the changes.
And it would have allowed me to cherry-pick the validated commits that don't need any changes, while reducing the code base that I really have to spend time on.
Here I'm almost forced to do a squash and work on it :p


Example: the removal of the code duplication was justified in the xiaomi_query file:
https://github.com/ysard/mi_remote_database/pull/3/commits/d576b005ee1d4220f9c72f51889561b9d97ac816

but the other changes are not because you had to undo them in:
https://github.com/ysard/mi_remote_database/pull/3/commits/eb907fafd0bc4744657f34b0587ef7320d1a2c83
and 
https://github.com/ysard/mi_remote_database/pull/3/commits/736499b08b49a3f2a8e0e30d8729fa2570db802a


Same for the deleted prints in __main__:
https://github.com/ysard/mi_remote_database/pull/3/commits/440309e9918beb929c540ce6ad296335077e4d9d
and restored:
https://github.com/ysard/mi_remote_database/pull/3/commits/736499b08b49a3f2a8e0e30d8729fa2570db802a


The migration of tvkill_export in a separated file is ok but should have been the subject of an independent commit.

Anyway it's too late and I think you got the idea.

Note: remarks are not meant to be harsh :D

File: src/xiaomi_parser.py [hunk 1]
```diff
@@ -374,11 +367,15 @@ def build_patterns(models):
         if "ir_zip_key_r" in model:
             # Decrypt IR code
             ir_code = process_xiaomi_shit(model["ir_zip_key_r"])
+            button_name = model.get("id", None)
```
Comment (ysard) [inline src/xiaomi_parser.py:370]: - None is already a default val
- Understood the need to keep the name of the button 
- Will rename id to something more explicit 'key_name'
- build_patterns was made to build patterns from models in json brands not models in models/ dir.
Will do a separated func for that

**Comment by ysard** [comment]
I close this issue, the functionality seems to be implemented now.
Thanks for the implementation example :)

**Pull request closed without merging**